native-activity/app/src/main/cpp/main.cpp

```diff
@@ -358,18 +404,14 @@ void android_main(android_app* state) {
     engine.state = *(SavedState*)state->savedState;
   }
 
-  // loop waiting for stuff to do.
-
   while (true) {
     // Read all pending events.
     int events;
     android_poll_source* source;
 
-    // If not animating_, we will block forever waiting for events.
-    // If animating_, we loop until all events are read, then continue
-    // to draw the next frame of animation.
-    while ((ALooper_pollAll(engine.animating() ? 0 : -1, nullptr, &events,
-                            (void**)&source)) >= 0) {
+    // Our input, sensor, and update/render logic is all driven by callbacks, so
+    // we don't need to use the non-blocking poll.
+    while ((ALooper_pollAll(-1, nullptr, &events, (void**)&source)) >= 0) {
```

Comment:
use pollOnce instead?

Reply:
That's the next PR. Will upload once this merges (github won't let me stack reviews like gerrit does)